### ISSUE TITLE
Scheduled Transaction body differs

### ISSUE THREAD

**Issue by failfmi** [hashgraph/hedera-sdk-go#314]
### Description

When 3 separate clients execute a scheduled transaction, which has the same schedulable transaction body inside, it turns out, that the sdk somehow shifts the transfers and resolves to 2+ more different schedule ids, because the inner transaction body is different.

### Steps to reproduce

The scripts showcases the creation of a 1:1 identical scheduled transaction, executed by 3 different clients. A total of 3 tinybars is sent from a threshold account (2/3), to which each of the clients' private key is part of the threshold account.

Expected behaviour should be the second and third transaction to resolve with `IDENTICAL_SCHEDULE_ALREADY_CREATED`.

`Note!`: It does not occur every time..

```go
package main

import (
	"fmt"
	"github.com/hashgraph/hedera-sdk-go/v2"
	"time"
)

func main() {
	accountId1, _ := hedera.AccountIDFromString("")
	privateKey1, _ := hedera.PrivateKeyFromString("")

	client1 := hedera.ClientForTestnet()
	client1 = client1.SetOperator(accountId1, privateKey1)

	accountId2, _ := hedera.AccountIDFromString("")
	privateKey2, _ := hedera.PrivateKeyFromString("")

	client2 := hedera.ClientForTestnet()
	client2 = client2.SetOperator(accountId2, privateKey2)

	accountId3, _ := hedera.AccountIDFromString("")
	privateKey3, _ := hedera.PrivateKeyFromString("")

	client3 := hedera.ClientForTestnet()
	client3 = client3.SetOperator(accountId3, privateKey3)

	thresholdAccount, _ := hedera.AccountIDFromString("")

	receivers := []hedera.AccountID{
		accountId1,
		accountId2,
		accountId3,
	}

	amount := int64(3) // Distributes to all 3 of them 1 tiny bar

	memo := time.Now().UTC().String()

	sendScheduledCreate(client1, thresholdAccount, receivers, amount, memo)
	sendScheduledCreate(client2, thresholdAccount, receivers, amount, memo)
	sendScheduledCreate(client3, thresholdAccount, receivers, amount, memo)
}

func sendScheduledCreate(client *hedera.Client, sender hedera.AccountID, toAccounts []hedera.AccountID, amount int64, memo string) {
	accountsLen := int64(len(toAccounts))
	perAccount := amount / accountsLen

	tx := hedera.NewTransferTransaction()

	for _, account := range toAccounts {
		tx.AddHbarTransfer(account, hedera.HbarFromTinybar(perAccount))
	}
	tx.AddHbarTransfer(sender, hedera.HbarFromTinybar(amount).Negated())

	tx, err := tx.FreezeWith(client)
	if err != nil {
		panic(err)
	}

	signedTransaction, err := tx.SignWithOperator(client)
	if err != nil {
		panic(err)
	}

	scheduledTx, err := hedera.NewScheduleCreateTransaction().
		SetScheduledTransaction(signedTransaction)

	if err != nil {
		panic(err)
	}
	scheduledTx = scheduledTx.
		SetPayerAccountID(sender).
		SetScheduleMemo(memo)

	response, err := scheduledTx.Execute(client)
	if err != nil {
		panic(err)
	}

	receipt, err := response.GetReceipt(client)
	if err != nil {
		panic(err)
	}
	fmt.Println("ScheduleID : ", receipt.ScheduleID)
}
```

### Additional context

![image](https://user-images.githubusercontent.com/20928965/141155266-4ca19f70-1b67-4406-bac0-51a9c891c1c2.png)

When you query schedule ids `0.0.7863617`, `0.0.7863618`, `0.0.7863620` and parse their transaction bodies
`CIDC1y9KLgosCgkKBRjl6qoBEAIKCQoFGObqqgEQAgoJCgUY5+qqARACCgkKBRjp6qoBEAU=`, `CIDC1y9KLgosCgkKBRjn6qoBEAIKCQoFGOnqqgEQBQoJCgUY5eqqARACCgkKBRjm6qoBEAI=`
and `CIDC1y9KLgosCgkKBRjp6qoBEAUKCQoFGOXqqgEQAgoJCgUY5uqqARACCgkKBRjn6qoBEAI=`
you can see that the bodies differ:
```
accountAmounts:{accountID:{accountNum:2798949} amount:1} accountAmounts:{accountID:{accountNum:2798950} amount:1} accountAmounts:{accountID:{accountNum:2798951} amount:1} accountAmounts:{accountID:{accountNum:2798953} amount:-3}
accountAmounts:{accountID:{accountNum:2798951} amount:1} accountAmounts:{accountID:{accountNum:2798953} amount:-3} accountAmounts:{accountID:{accountNum:2798949} amount:1} accountAmounts:{accountID:{accountNum:2798950} amount:1}
accountAmounts:{accountID:{accountNum:2798953} amount:-3} accountAmounts:{accountID:{accountNum:2798949} amount:1} accountAmounts:{accountID:{accountNum:2798950} amount:1} accountAmounts:{accountID:{accountNum:2798951} amount:1}
```

Code to parse the bodies:

```go
package main

import (
	"encoding/base64"
	"fmt"
	"github.com/hashgraph/hedera-sdk-go/v2/proto"
	proto2 "google.golang.org/protobuf/proto"
)

func main() {
	txBody1 := "CIDC1y9KLgosCgkKBRjl6qoBEAIKCQoFGObqqgEQAgoJCgUY5+qqARACCgkKBRjp6qoBEAU="
	parseBody(txBody1)

	txBody2 := "CIDC1y9KLgosCgkKBRjn6qoBEAIKCQoFGOnqqgEQBQoJCgUY5eqqARACCgkKBRjm6qoBEAI="
	parseBody(txBody2)

	txBody3 := "CIDC1y9KLgosCgkKBRjp6qoBEAUKCQoFGOXqqgEQAgoJCgUY5uqqARACCgkKBRjn6qoBEAI="
	parseBody(txBody3)
}

func parseBody(body string) {
	bytes, err := base64Decode([]byte(body))
	if err != nil {
		panic(err)
	}

	schedulableTxBody :=proto.SchedulableTransactionBody{}

	err = proto2.Unmarshal(bytes, &schedulableTxBody)
	if err != nil {
		panic(err)
	}
	fmt.Println(schedulableTxBody.GetCryptoTransfer().Transfers)
}

func base64Decode(message []byte) (b []byte, err error) {
	var l int
	b = make([]byte, base64.StdEncoding.DecodedLen(len(message)))
	l, err = base64.StdEncoding.Decode(b, message)
	if err != nil {
		return
	}
	return b[:l], nil
}
```

### Hedera network

testnet

### Version

v2.4.0

### Operating system

macOS

**Comment by danielakhterov** [comment]
From a quick skim, I'm going to bet this is the result of us using a `map` to store the transfers. When we iterate over the map I'm _assuming_ the order is not guaranteed which results is different transfer lists. Definitely should be fixed, but I'm a bit confused as to why this isn't an issue in Java or JS (or maybe it is and we haven't noticed it yet)

**Comment by gregscullard** [comment]
I would guess that in testing we only supply a single transfer which means the map only contains one item. The issue may therefore exist on the other SDKs.

**Comment by danielakhterov** [comment]
> I would guess that in testing we only supply a single transfer which means the map only contains one item. The issue may therefore exist on the other SDKs.

We should definitely add a unit test that serializes the transfers the same way every time. 

We can probably achieve this by simply sorting the list of transfers by account ID. That or perhaps we should use a list to store the transfers internally and only cover them into a map when someone calls `GetHbarTransfers()`

**Comment by danielakhterov** [comment]
I believe this has now been fixed by #315 and has been released https://github.com/hashgraph/hedera-sdk-go/releases/tag/v2.4.1
